ext/posix/posix.c

```diff
@@ -453,14 +453,16 @@ PHP_FUNCTION(posix_ttyname)
 		Z_PARAM_ZVAL(z_fd)
 	ZEND_PARSE_PARAMETERS_END();
 
-	switch (Z_TYPE_P(z_fd)) {
-		case IS_RESOURCE:
-			if (!php_posix_stream_get_fd(z_fd, &fd)) {
-				RETURN_FALSE;
-			}
-			break;
-		default:
+	if (Z_TYPE_P(z_fd) == IS_RESOURCE) {
+		if (!php_posix_stream_get_fd(z_fd, &fd)) {
+			RETURN_FALSE;
+		}
+	} else {
+		if (!zend_parse_arg_long(z_fd, &fd, /* is_null */ false, /* check_null */ false, /* arg_num */ 1)) {
+			php_error_docref(NULL, E_WARNING, "Argument #1 ($file_descriptor) must be of type int|resource, %s given",
+				zend_zval_type_name(z_fd));
 			fd = zval_get_long(z_fd);
+		}
```

Comment:
This makes perfect sense to me, but we probably should check that the `fd <= INT_MAX` (since we're later passing the value to a function which only expects `int`). Well, we didn't check that so far, so it would be a hypothetical BC break.

Reply:
Right, and we probably should do it below as well.
Currently, the value would just have been truncated, correct? So it was kinda always broken, maybe just add another warning?

Reply:
> Currently, the value would just have been truncated, correct?

Most likely, but according to the specs the behavior is implementation-defined.

So, yeah, a warning seems to be a good compromise between BC break and proper behavior (`ValueError`). And that should also be done in `posix_ttyname()`.